cpp/src/io/avro/avro_gpu.cu

```diff
@@ -20,8 +20,8 @@ namespace cudf {
 namespace io {
 namespace avro {
 namespace gpu {
-#define NWARPS 16
-#define MAX_SHARED_SCHEMA_LEN 1000
+constexpr int nwarps                     = 16;
+constexpr uint32_t max_shared_schema_len = 1000;
```

Comment:
Instead of making these global variables, they should be non-type template parameters to the `gpuDecodeAvroColumnData` kernel.